source/NVDAObjects/IAccessible/winword.py

```diff
@@ -369,6 +377,36 @@ def script_previousParagraph(self,gesture):
 		self._caretScriptPostMovedHelper(textInfos.UNIT_PARAGRAPH,gesture,None)
 	script_previousParagraph.resumeSayAllMode = sayAll.CURSOR.CARET
 
+	@script(
+		gestures=(
+			"kb:control+v",
+			"kb:control+x",
+			"kb:control+y",
+			"kb:control+z",
+			"kb:alt+backspace",
+			"kb:backspace",
+			"kb:control+backspace",
+		)
+	)
+	def script_updateBrailleAndReviewPosition(self, gesture: inputCore.InputGesture) -> None:
+		"""Helper script to update braille and review position.
+		Caret event is not always fired when control+v, control+x, control+y
+		or control+z (alt+backspace) is pressed.
+		When backspace or control+backspace is pressed and hold down, braille
+		may not always be updated. Allowing braille and review position updates
+		in L{event_caret} should fix that problem.
+		"""
+		# Ensuring braille and review position updates are allowed in caret event.
+		self._fromUpdateBrailleAndReviewPosition = True
+		# Speech output when backspace or control+backspace is pressed.
+		if gesture.displayName == "backspace":

```

Comment:
Rather than matching on key names I'd suggest to override `_backspaceScriptHelper` for this class, and handle only remaining combinations inside this script. Note that `_backspaceScriptHelper` is executed for both `backspace` and `ctrl_backspace`.

Reply:
Sorry but I do not understand why. If `_backspaceScriptHelper` should be changed sometimes then it should likely be changed within this class as well.

Reply:
To reiterate on my last comment. Currently you have created a single script which is bound to multiple key combinations. Depending on what key presses invoked the script it behaves differently (there is a common part - setting the class variable, but it is quite small). Since the differences in behavior can be moved to separate methods inside the class that is always better than adding multiple conditionals into a single one. All commands invoked with backslash eventually execute `_backspaceScriptHelper`, so it makes sense to override it for the class, set the class variable to `True` inside it and call the superclass method. That way your script is only concerned with the remaining gestures, so it does not need to handle calling other scripts (they should mostly not be called  programmatically, rather in response to user input). Is this making more sense for you now?